### PR TITLE
Merge lib with storybook

### LINKED ISSUE #69: Move storybook to same project as package
Often times, the storybook app is integrated directly into the package project, allowing one to write `Component.stories.tsx` files alongside the `Component.tsx` file. This has the advantage of co-locating stories with their respective components (or in our case hooks).

[This component library template](https://github.com/IgnacioNMiranda/vite-component-library-template) is similar to our library in the tech it uses, though the repo is structured a bit differently. Among the differences includes merging the storybook and package projects.

### PR DESCRIPTION
**Pull request by PatrickDeVries**
### Why:

Closes #69 (question answered by doing it)

### What's being changed (if available, include any code snippets, screenshots, or gifs):

- Colocate library and storybook code in folders and move it all to the root directory
  - Update stories to import hooks directly from their files
  - Update mdx to import types from root dist .d.ts files
- Merge ESLint, Prettier, and TS configs
- Merge package.jsons
- Update library `vite.config.ts` to ignore react-dom and storybook files
- Move storybook `vite.config.ts` into .storybook
- Move development README.md content into CONTRIBUTING.md
- Update paths/run conditions in all workflows

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,93 @@
...
+    ".": {
+      "types": "./dist/index.d.ts",
+      "import": "./dist/too-many-hooks.js",
+      "require": "./dist/too-many-hooks.umd.cjs"
+    }
+  },
+  "scripts": {
+    "build": "tsc && vite build",
+    "storybook": "storybook dev -p 6006",
+    "build-storybook": "storybook build",
```
Comment (dawsonbooth) [inline package.json:31]: Now that the projects are consolidated, should we run the `build` script as part of these scripts so we always get the latest code updates?

```suggestion
    "storybook": "bun run build && storybook dev -p 6006",
    "build-storybook": "bun run build && storybook build",
```

File: tsconfig.eslint.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "include": ["src", "vite.config.ts", ".storybook/**.ts", "tailwind.config.ts"]
```
Comment (dawsonbooth) [inline tsconfig.eslint.json:3]: If this still works it's a little simpler

```suggestion
  "include": ["src", "vite.config.ts", ".storybook", "tailwind.config.ts"]
```

**Review by dawsonbooth — CHANGES_REQUESTED** [review]

**Review by dawsonbooth — CHANGES_REQUESTED** [review]
Not sure if I like `src/common` being used just for storybook components and assets.

Maybe those should be moved to a folder called `storybook-common` or something more clearly indicating that they're just used in stories.

File: tsconfig.eslint.json (hunk 2 above)
Comment (PatrickDeVries) [reply tsconfig.eslint.json:3]: Unfortunately it does not work
image

**Review by dawsonbooth — APPROVED** [review]

**Pull request merged**